src/corefx/System.Globalization.Native/pal_localeNumberData.c

```diff
@@ -169,35 +160,46 @@ int GetNumericPattern(const UNumberFormat* pNumberFormat, const char* patterns[]
     const int MAX_DOTNET_NUMERIC_PATTERN_LENGTH = 6; // example: "(C n)" plus terminator
 
     UErrorCode ignore = U_ZERO_ERROR;
-    int32_t icuPatternLength = unum_toPattern(pNumberFormat, false, nullptr, 0, &ignore);
+    int32_t icuPatternLength = unum_toPattern(pNumberFormat, false, NULL, 0, &ignore) + 1;
 
-    std::vector<UChar> icuPattern(icuPatternLength + 1, '\0');
+    UChar* icuPattern = calloc(icuPatternLength, sizeof(UChar));
+    if (icuPattern == NULL)
+    {
+        return U_MEMORY_ALLOCATION_ERROR;
+    }
 
     UErrorCode err = U_ZERO_ERROR;
 
-    unum_toPattern(pNumberFormat, false, icuPattern.data(), icuPattern.size(), &err);
+    unum_toPattern(pNumberFormat, false, icuPattern, icuPatternLength, &err);
 
     assert(U_SUCCESS(err));
 
-    std::string normalizedPattern = NormalizeNumericPattern(icuPattern.data(), isNegative);
+    char* normalizedPattern = NormalizeNumericPattern(icuPattern, isNegative);
```

Comment:
>NormalizeNumericPattern [](start = 30, length = 23)

Maybe we can male this method return the length as we already know it there and avoid calling strlen after

Reply:
Yes, I will do that.